### PR TITLE
[Android] Support isUrgent variable in Event Path

### PR DESCRIPTION
**Pull request by joonhaengHeo**
#### Problem
What is being fixed? 
issue #22526
* Matter spec is defined "isUrgent" variable in EventPathIB.
* https://github.com/CHIP-Specifications/connectedhomeip-spec/blob/master/src/data_model/Encoding-Specification.adoc#encoding-EventPathIB
* It can be supported in Android Platform.

#### Change overview
Add isUrgent in EventPath class.

#### Testing
How was this tested? (at least one bullet point required)
* Commissioning Test in Android platform.
* Send Event Path (isUrgent variable is true and false).

### CODE REVIEW

File: src/app/MessageDef/EventDataIB.cpp [hunk 1]
```diff
@@ -242,6 +242,8 @@ CHIP_ERROR EventDataIB::Parser::ProcessEventPath(EventPathIB::Parser & aEventPat
     err = aEventPath.GetEvent(&(aConcreteEventPath.mEventId));
     VerifyOrReturnError(err == CHIP_NO_ERROR, CHIP_ERROR_IM_MALFORMED_EVENT_PATH_IB);
 
+    err = aEventPath.GetIsUrgent(&(aConcreteEventPath.mIsUrgentEvent));
```
Comment (bzbarsky-apple) [inline src/app/MessageDef/EventDataIB.cpp:245]: Why do we want to do this?  The spec is not very clear on this, but IsUrgent is an input, not an output; EventDataIB should never have it present...

Filed https://github.com/CHIP-Specifications/connectedhomeip-spec/issues/5596 to get this clarified.

**Review by bzbarsky-apple — CHANGES_REQUESTED** [review]

File: src/app/EventManagement.cpp [hunk 2]
```diff
@@ -289,6 +289,7 @@ CHIP_ERROR EventManagement::ConstructEvent(EventLoadOutContext * apContext, Even
     eventPathBuilder.Endpoint(apOptions->mPath.mEndpointId)
         .Cluster(apOptions->mPath.mClusterId)
         .Event(apOptions->mPath.mEventId)
+        .IsUrgent(apOptions->mPath.mIsUrgentEvent)
```
[the remark this answers is not in the document]
Comment (yunhanw-google) [reply src/app/EventManagement.cpp:292]: isUrgent is used for requested path for subscribe, and should not be used in event Path builder when constructing event in event management.

**Review by yunhanw-google — CHANGES_REQUESTED** [review]
isUrgent is used for requested path for subscribe, and should not be used in event Path builder when constructing event in event management.

File: src/app/MessageDef/EventDataIB.cpp (hunk 1 above)
Comment (yunhanw-google) [reply src/app/MessageDef/EventDataIB.cpp:245]: agree, it should be input

File: src/controller/java/CHIPDeviceController-JNI.cpp [hunk 3]
```diff
@@ -1326,7 +1332,7 @@ CHIP_ERROR ParseEventPath(jobject eventPath, EndpointId & outEndpointId, Cluster
     outEndpointId = static_cast<EndpointId>(endpointId);
     outClusterId  = static_cast<ClusterId>(clusterId);
     outEventId    = static_cast<EventId>(eventId);
-
+    outIsUrgentId = isUrgentEvent;
```
Comment (yunhanw-google) [inline src/controller/java/CHIPDeviceController-JNI.cpp:1335]: there is no need to parse the urgent flag when reading event from report.

**Comment by yunhanw-google** [comment]
as I point out in this PR, you should just modify java event path, https://github.com/project-chip/connectedhomeip/pull/20198#discussion_r918554666, then modify subscribe request with optional urgent event path in WildcardFragment,

+++ b/src/android/CHIPTool/app/src/main/java/com/google/chip/chiptool/clusterclient/WildcardFragment.kt
@@ -27,6 +27,7 @@ import kotlinx.android.synthetic.main.wildcard_fragment.attributeIdEd
 import kotlinx.android.synthetic.main.wildcard_fragment.clusterIdEd
 import kotlinx.android.synthetic.main.wildcard_fragment.endpointIdEd
 import kotlinx.android.synthetic.main.wildcard_fragment.eventIdEd
+import kotlinx.android.synthetic.main.wildcard_fragment.urgentEd
 import kotlinx.android.synthetic.main.wildcard_fragment.outputTv
 import kotlinx.android.synthetic.main.wildcard_fragment.view.readBtn
 import kotlinx.android.synthetic.main.wildcard_fragment.view.readEventBtn
@@ -133,6 +134,7 @@ class WildcardFragment : Fragment() {
     val clusterId = getChipPathIdForText(clusterIdEd.text.toString())
     val attributeId = getChipPathIdForText(attributeIdEd.text.toString())
     val eventId = getChipPathIdForText(eventIdEd.text.toString())
+    val urgent = Boolean.valueOf(urgentEd.text.toString())
 
     if (type == ATTRIBUTE) {
       val attributePath = ChipAttributePath.newInstance(endpointId, clusterId, attributeId)
@@ -144,7 +146,7 @@ class WildcardFragment : Fragment() {
                                        minInterval,
                                        maxInterval)
     } else if (type == EVENT) {
-      val eventPath = ChipEventPath.newInstance(endpointId, clusterId, eventId)
+      val eventPath = ChipEventPath.newInstance(endpointId, clusterId, eventId, urgent)
       deviceController.subscribeToEventPath(subscriptionEstablishedCallback,
                                       resubscriptionAttemptCallback,
                                       reportEventCallback,
@@ -169,7 +171,7 @@ class WildcardFragment : Fragment() {
                           addressUpdateFragment.deviceId),
                           listOf(attributePath))
     } else if (type == EVENT) {
-      val eventPath = ChipEventPath.newInstance(endpointId, clusterId, eventId)
+      val eventPath = ChipEventPath.newInstance(endpointId, clusterId, eventId, false)
       deviceController.readEventPath(reportEventCallback,
                           ChipClient.getConnectedDevicePointer(requireContext(),
                           addressUpdateFragment.deviceId),

**Comment by joonhaengHeo** [comment]
I will check comments and spec.

**Pull request closed without merging**